### ISSUE TITLE
Ical4j 4.0.1 introduced duplicated classes

### ISSUE THREAD

**Issue by johandorland** [ical4j/ical4j#708]
**Describe the bug**
We use the Maven enforcer plugin to check our dependencies, of which ical4j is one.
With the update to 4.0.1 the enforcer plugin failed with the following message. It seems like ical4j itself is packaging duplicate classes. 
This issue did not exist in 4.0.0, so it seems like it was introduced in 4.0.1.

```sh
[ERROR] Rule 0: org.codehaus.mojo.extraenforcer.dependencies.BanDuplicateClasses failed with message:
[ERROR] Duplicate classes found:
[ERROR] 
[ERROR]   Found in:
[ERROR]     org.mnode.ical4j:ical4j:jar:4.0.1:compile
[ERROR]   Duplicate classes:
[ERROR]     net/fortuna/ical4j/model/parameter/Display.class
[ERROR]     net/fortuna/ical4j/validate/PropertyRuleSet$1.class
[ERROR]     net/fortuna/ical4j/model/property/Categories.class
[ERROR]     net/fortuna/ical4j/model/property/TzUntil$Factory.class
[ERROR]     net/fortuna/ical4j/model/property/Summary.class
[ERROR]     net/fortuna/ical4j/model/property/immutable/ImmutableBusyType.class
[ERROR]     net/fortuna/ical4j/model/property/StreetAddress.class
.........
```

**To Reproduce**
I added a simplified maven `pom.xml` down below
1. Create a maven project with ical4j and use the enforcer plugin
2. Run `mvn enforcer:enforce`
3. See error

When building the 4.0.1 tag of ical4j with `./gradlew clean  build -x test`, I see gradle is throwing the following warnings, which are not present in 4.0.0.
```
Encountered duplicate path "net/fortuna/ical4j/data/CalendarBuilder.class" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "net/fortuna/ical4j/data/CalendarParser.class" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "net/fortuna/ical4j/data/CalendarParserFactory.class" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "net/fortuna/ical4j/data/CalendarParserImpl$ComponentListParser.class" during copy operation configured with DuplicatesStrategy.WARN
Encountered duplicate path "net/fortuna/ical4j/data/CalendarParserImpl$ComponentParser.class" during copy operation configured with DuplicatesStrategy.WARN
``` 

**Expected behavior**
I would expect ical4j to not package duplicate classes.

**Environment (please complete the following information):**
 - OS: Linux
 - Java Version: 17
 - iCal4j Version: 4.0.1

**Additional context**
Apparently the duplicates are perfect copies, because when the `ignoreWhenIdentical` option is enabled, it works.
```xml
<project>
    <modelVersion>4.0.0</modelVersion>

    <groupId>com.mycompany.app</groupId>
    <artifactId>my-app</artifactId>
    <version>1</version>

    <dependencies>
        <dependency>
            <groupId>org.mnode.ical4j</groupId>
            <artifactId>ical4j</artifactId>
            <version>4.0.1</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-enforcer-plugin</artifactId>
                <version>3.5.0</version>
                <configuration>
                    <rules>
                        <banDuplicateClasses>
                            <findAllDuplicates>true</findAllDuplicates>
                            <!-- <ignoreWhenIdentical>true</ignoreWhenIdentical> -->
                        </banDuplicateClasses>
                    </rules>
                    <fail>true</fail>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.codehaus.mojo</groupId>
                        <artifactId>extra-enforcer-rules</artifactId>
                        <version>1.8.0</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
</project>
```

**Comment by girgen** [comment]
When using ical4j in a tomcat 11 container, we get 

``` main ERROR Unable to create Lookup for bundle java.lang.ClassCastException: class org.apache.logging.log4j.core.lookup.ResourceBundleLookup```

so the distribution of ical4j past version 4.0.0 is unusable. It's a bit worrying that nothing has been done about this for a number of patch releases.

**Comment by girgen** [comment]
This i fixed in version 4.0.6, it seems.

**Comment by girgen** [comment]
My bad, it is still a problem in 4.0.6. We are forced to download it manually, jar x + jar c to get a correct jar file.

**Comment by benfortuna** [comment]
Thanks, I suspect it may be caused by the module plugin, but haven't been able to track down exactly why. I will focus on resolution before next release.

**Comment by benfortuna** [comment]
This is now fixed in the latest 4.0.7 release
